swift/141-Linked-List-Cycle.swift

```diff
@@ -0,0 +1,31 @@
+/**
+ * Definition for singly-linked list.
+ * public class ListNode {
+ *     public var val: Int
+ *     public var next: ListNode?
+ *     public init(_ val: Int) {
+ *         self.val = val
+ *         self.next = nil
+ *     }
+ * }
+ */
+
+class Solution {
+    func hasCycle(_ head: ListNode?) -> Bool {
+        if head === nil {
+            return false
+        }
+        
```

Comment:
Actually you can catch the optional, so that in subsequent lines, you don't need to use chaining optionals.
```suggestion
        guard let head = head else {
            return false
        }
        
        // In subsequent lines, simply can use
        fast = fast.next
        slow = slow.next
```

Reply:
Thank you for the suggestion. I looked at it but in a while loop, there's the issue of unwrapping the ListNode type. I think the optionals are necessary because based on Leetcode's definition of Linked List for swift, the next node in a Linked List is an optional ListNode type since the next node can be null.